### ISSUE TITLE
*.Rproj is created for new project, but "Open project" dialog allows to show lowercase *.rproj only

### ISSUE THREAD

**Issue by beerda** [rstudio/rstudio#15340]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2024.09.0 Build 375
    OS Version      :  Ubuntu 22.04
    R Version       :  4.4.1

### Steps to reproduce the problem

1. Create new project -  `*.Rproj` file is created
2. Open existing project and browse to the newly created project directory
3. `*.Rproj` file cannot be seen.
4. Rename `*.Rproj` manually to `*.rproj`
5. `*.rproj` file can be seen in the "open project" dialog

### Describe the problem in detail

Open project dialog does not allow to open a project file that has "Rproj" extension. The extension should be manually renamed to "rproj" in order to open the project. On the other hand, New project dialog creates project file with "Rproj" and not "rproj".

### Describe the behavior you expected

I expect both "Rproj" and "rproj" be allowed in the open project dialog.

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
I stepped through the code and we are passing in a slightly incorrect filter: `R Projects (*.RProj)` (notice the capital P in RProj).

However, even when I adjusted this to `*.Rproj` it doesn't work. I suspect this is an Electron issue (it seems to be internally lowercasing the extensions). I'm going to confirm and open a bug against Electron.

In the meantime, one option is to switch to the same built-in dialogs that RStudio Server uses. 

That's in Tools / Global Options / General / Advanced: uncheck "Use native file and message dialog boxes"

![Image](https://github.com/user-attachments/assets/98a1c556-0cbe-44b0-a067-ec4c61b106a4)

**Comment by gtritchie** [comment]
Confirmed this is an Electron issue: https://github.com/electron/electron/issues/44398

Perhaps as an escape hatch, we should include the "All files *" filter to at least give the option to see everything in the folder?

**Comment by gtritchie** [comment]
Yup, we had switched to the native dialogs in Kousa Dogwood, but this issue has prompted us to change the default back to the web based dialogs. Maybe next time...

**Comment by goebbe** [comment]
Has this issue been resolved? 
I was testing the optional "use native file and message dialogue boxes" on Linux Mint, Cinnamon and had no problems opening the *.Rproj files using the GTK-based, native dialogue boxes on Linux.  
Rstudio: 2025.09.1 Build 401
LinuxMint 22.2  Nemo 6.4.5

**Comment by beerda** [comment]
Now it seems to work well again in RStudio 2025.09.1 Build 401

**Comment by gtritchie** [comment]
Reopening, want our QA team to do additional validation.

**Comment by gtritchie** [comment]
This was fixed in Electron. Last repro'd in 2024.12, has been fixed in 2025.05 and newer.

**Comment by kevinushey** [comment]
Confirmed fixed in the latest daily builds -- 2025.11.0+263; tested on Ubuntu 24.04.
